src/plugins/network.ts

```diff
@@ -23,336 +23,217 @@
-        this._onLoadingFinished(data);
-      } else {
-        /**
-         * Edge case, we handle it to proceed with next navigation
-         */
-        this._onLoadingFailed(
-          Object.assign(event, {
-            type: event.type,
-            errorText: 'redirectResponse data is not available',
-          })
-        );
-      }
-      const redirectedRecord = this._findNetworkRecord(requestId);
-      /**
-       * Rewrite the map with new redirect id to not reset
-       * the redirect request with original request
-       */
-      this._InflightRequestMap.delete(requestId);
-      this._InflightRequestMap.set(
-        `redirect:${timestamp}:${requestId}`,
-        redirectedRecord
-      );
-    }
 
-    this._InflightRequestMap.set(requestId, {
+    const timestamp = getTimestamp();
+    const networkEntry: NetworkInfo = {
       browser: this._browser,
       step: this._currentStep,
-      timestamp: getTimestamp(),
+      timestamp,
       url,
-      request,
-      type,
-      method,
-      requestSentTime: timestamp,
-      isNavigationRequest,
-      status: 0,
+      type: request.resourceType(),
+      method: request.method(),
+      requestSentTime: epochTimeInMills(),
+      request: {
+        url,
+        method: request.method(),
+        headers: {},
+      },
+      response: {
+        statusCode: -1,
+        mimeType: 'x-unknown',
+        headers: {},
+        redirectURL: '',
+      },
+      isNavigationRequest: request.isNavigationRequest(),
+      status: -1,
       loadEndTime: -1,
       responseReceivedTime: -1,
-      response: null,
-      resourceSize: 0,
-      transferSize: 0,
+      resourceSize: -1,
+      transferSize: -1,
       timings: null,
-    });
-  }
-
-  _onRequestWillBeSentExtraInfo(
-    event: Protocol.Network.requestWillBeSentExtraInfoPayload
-  ) {
-    const { requestId, headers } = event;
-    const record = this._findNetworkRecord(requestId);
-    if (!record) {
-      return;
-    }
-    /**
-     * Enhance request headers with additional information
-     */
-    record.request.headers = {
-      ...record.request.headers,
-      ...headers,
     };
-  }
 
-  _onDataReceived(event: Protocol.Network.dataReceivedPayload) {
-    const { requestId, dataLength, encodedDataLength } = event;
-    const record = this._findNetworkRecord(requestId);
-    if (!record) {
-      return;
+    if (request.redirectedFrom()) {
+      const fromEntry = this._findNetworkEntry(request.redirectedFrom());
+      if (fromEntry) fromEntry.response.redirectURL = request.url();
     }
-    record.resourceSize += dataLength;
-    if (encodedDataLength >= 0) record.transferSize += encodedDataLength;
-  }
-
-  _onResponseReceived(event: Protocol.Network.responseReceivedPayload) {
-    const { requestId, response, timestamp } = event;
-    const record = this._findNetworkRecord(requestId);
-    if (!record) {
-      return;
+    (request as any)[this._networkEntrySymbol] = networkEntry;
```

Comment:
Could you extend the request type to prevent having to cast it to `any`?
I'm playing around with this on my local copy of your PR, and it seems to be working okay. 

```
export const networkEntrySymbol = Symbol.for('networkEntry');

export type NetworkRequest = Request & {
  [networkEntrySymbol]?: NetworkInfo
}
```